src/ot_orchestration/utils/dataproc_platform.py

```diff
@@ -0,0 +1,225 @@
+    DataprocCreateClusterOperator as _DataprocCreateClusterOperator,
+)
+from airflow.providers.google.cloud.operators.dataproc import (
+    DataprocSubmitJobOperator as _DataprocSubmitJobOperator,
+)
+from google.cloud.dataproc_v1 import Cluster, JobReference
+from google.cloud.dataproc_v1.types.jobs import Job, JobPlacement, SparkJob
+
+from ot_orchestration.utils import random_id
+from ot_orchestration.utils.common import GCP_PROJECT_PLATFORM, GCP_REGION
+from ot_orchestration.utils.dataproc import ClusterGenerator
+from ot_orchestration.utils.labels import Labels
+
+
+class DataprocCreateClusterOperator(_DataprocCreateClusterOperator):
+    """Create a new Dataproc cluster tailored for running Platform ETL.
+
+    This class sets many default values and streamlines the cluster creation to
+    the needs of ETL. Refer to the parent class for details on extending this one.
+
+    For more information on how to use this operator, take a look at
+    `the guide <https://airflow.apache.org/docs/apache-airflow-providers-google/stable/operators/cloud/dataproc.html#howto-operator-dataproccreateclusteroperator>`_.
+
+    Args:
+        project_id: The ID of the Google cloud project in which
+            to create the cluster. (templated)
+        region: The specified region where the dataproc cluster is created.
+        cluster_name: Name of the cluster to create
+        cluster_config: Required. The cluster config to create.
+            If a dict is provided, it must be of the same form as the protobuf message
+            :class:`~google.cloud.dataproc_v1.types.ClusterConfig`
+        idle_delete_ttl: Optional. The duration to keep the cluster alive while idling.
+        labels: Labels that will be assigned to created cluster. Please, notice that
+            adding labels to ClusterConfig object in cluster_config parameter will not lead
+            to adding labels to the cluster. Labels for the clusters could be only set by passing
+            values to parameter of DataprocCreateCluster operator.
+        labels: Labels that will be assigned to created cluster.
+        metadata: Additional metadata that is provided to the method.
+        gcp_conn_id: The connection ID to use connecting to Google Cloud.
+        impersonation_chain: Optional service account to impersonate using short-term
+            credentials, or chained list of accounts required to get the access_token
+            of the last account in the list, which will be impersonated in the request.
+            If set as a string, the account must grant the originating account
+            the Service Account Token Creator IAM role.
+            If set as a sequence, the identities from the list must grant
+            Service Account Token Creator IAM role to the directly preceding identity, with first
+            account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields: Sequence[str] = (
```

Comment:
Do you need to overwrite the template_fields?

Reply:
I need to change them to specify the fields I want templates to work on. I don't overwrite them in the parent class though, the parent (`DataprocSubmitJobOperator`) will still have its own template_fields class property.